### PR TITLE
unhide errors

### PR DESCRIPTION
**Pull request by tralexa**

### CODE REVIEW

**Comment by fejta** [comment]
/test pull-test-infra-verify-file-perms
/test pull-test-infra-verify-github-spelling

Why have we held this one?

File: prow/spyglass/artifacts.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ func (s *Spyglass) ListArtifacts(src string) ([]string, error) {
 			break
 		}
 	}
+
+	if err != nil {
+		logrus.Warningf("Failed to list artifacts for prow job: %v", err)
```
Comment (stevekuznetsov) [inline prow/spyglass/artifacts.go]: I think we're doing this on purpose? Perhaps this should be a `Debugf`? `Warning` and `Error` levels are reserved for things that admins should be looking for and fixing and I'm not convinced a normal deployment of `deck` won't just spit this out all the time. It looks like the method is written to handle the log being missing on purpose.

File: prow/spyglass/gcsartifact_fetcher.go [hunk 2]
```diff
@@ -79,12 +79,12 @@ func fieldsForJob(src *gcsJobSource) logrus.Fields {
 func newGCSJobSource(src string) (*gcsJobSource, error) {
 	gcsURL, err := url.Parse(fmt.Sprintf("gs://%s", src))
 	if err != nil {
-		return &gcsJobSource{}, ErrCannotParseSource
+		return &gcsJobSource{}, err
```
Comment (stevekuznetsov) [inline prow/spyglass/gcsartifact_fetcher.go]: Are there downstream consumers of this? It's an exported method so perhaps if we want to expose why we can't parse the source, we want a wrapper:


```go
type ErrCannotParseSource struct {
    inner error
}

func (e ErrCannotParseSource) Error() string {
    return fmt.Sprintf("could not create job source from provided source: %v", e.inner)
}

func IsErrCannotParseSource(err error) bool {
    if err == nil {
        return false
    }

   _, isParseErr := err.(*ErrCannotParseSource)
   return isParseErr
}
```

**Pull request closed without merging**